proxy.js

```diff
@@ -16,13 +16,22 @@ var port = settings.port;
 var app = connect();
 var server = http.createServer(app);
 
+var MS_TIMEOUT = process.env.MS_TIMEOUT || 7000;
+var MAX_REQUESTS = process.env.MAX_REQUESTS || 30;
+
 var count = 0;
 
 app.use(compression())
 .use(function (req, resp) {
   count += 1;
   console.log('info active_requests=' + count);
 
+  if (count > MAX_REQUESTS) {
+    resp.statusCode = 503;
+    resp.end('Server under high load, try again later');
+    return;
```

Comment:
Ah of course. I've moved this below the handler that counts down and it works properly now. I added a test for this, but it seems like Supertest isn't starting a new server for me with every test, so I have it skipped for now. If run individually, it passes.